internal/vulnerability/runner.go

```diff
@@ -70,3 +77,30 @@ func NewVulnerabilityRunner(log *logp.Logger) (VulnerabilityRunner, error) {
 func (f VulnerabilityRunner) GetRunner() artifact.Runner {
 	return f.runner
 }
+
+func clearTrivyCache(ctx context.Context, log *logp.Logger) error {
+	log.Info("Starting VulnerabilityRunner.ClearCache")
+	options := []flag.Options{
+		{CacheOptions: flag.CacheOptions{ClearCache: true}},
+		{DBOptions: flag.DBOptions{Reset: true}},
+	}
+
+	errs := make([]error, 0, len(options))
+
+	for _, opts := range options {
+		r, err := artifact.NewRunner(ctx, opts)
+		if err != nil {
+			if !errors.Is(err, artifact.SkipScan) {
+				errs = append(errs, err)
+			}
+		}
+
+		// it should never go here (NewRunner should always return artifact.SkipScan and nil runner), but just in case it goes, lets close the runner.
```

Comment:
If skip scan is always expected, I would invert the flow to highlight as main flow.

So 

```
  if errors.Is(err, trivy.SkipScan) {
    continue
  }

  // handle error and close r here then 
```

Reply:
It was like that at first https://github.com/elastic/cloudbeat/pull/2168/commits/6bd63491eaacfd324c54b6ee94769b94df7cdd99

But then I wanted to be 100% sure that if a runner is returned, we should close it.